### PR TITLE
Cleaned up openstack resources if provision fails

### PR DESCRIPTION
**Pull request by yaso195**
Currently, we dont clean up the VM if provision fails during volume/
floating ip attachment. With this change, if anything fails during
provisioning, we clean up the resources.

@variadico @lilirui @zquestz

### CODE REVIEW

File: virtualmachine/openstack/util.go [hunk 1]
```diff
@@ -425,6 +425,17 @@ func createAndAttachVolume(vm *VM) error {
...
 	}
 
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		errDeleteVolume := volumes.Delete(bsClient, vol.ID).ExtractErr()
+		if errDeleteVolume != nil {
+			err = fmt.Errorf("%s %s", err, errDeleteVolume)
```
Comment (lilirui) [inline virtualmachine/openstack/util.go]: Nit: The error in defer function will be discarded.

**Comment by lilirui** [comment]
LGTM, please make the test green.

**Comment by variadico** [comment]
You're eating all the errors from the VM destroy calls.

```go
func foo() error {
  defer func() {
    // Nothing happens with err.
    err = destroy()
  }()
}
```

If you wanted to use that pattern, then you probably meant to use named returns.

```go
// Notice: "err error".
func foo() (err error) {
  defer func() {
    // err will be returned.
    err = destroy()
  }()
}

```

File: virtualmachine/openstack/util.go (hunk 1 above)
Comment (yaso195) [inline virtualmachine/openstack/util.go]: Done

**Pull request merged**